File: 0290/solution.ts
function wordPattern(pattern: string, s: string): boolean {
  const letters: string[] = pattern.split("");
  const words: string[] = s.split(" ");
  if (letters.length !== words.length) {
    return false;
  }
  const pMap: Map<string, string> = new Map();
  const sMap: Map<string, string> = new Map();

  for (let i = 0; i < letters.length; i++) {
    if (pMap.has(letters[i]) && pMap.get(letters[i]) !== words[i]) {
      return false;
    }
    pMap.set(letters[i], words[i]);

    if (sMap.has(words[i]) && sMap.get(words[i]) !== letters[i]) {
      return false;
    }
    sMap.set(words[i], letters[i]);
  }
  return true;
}

File: 0290/solution.go
package main

import (
	"strings"
)

func wordPattern(p string, str string) bool {
    s := strings.Split(str, " ")
    pMap := make(map[byte]string)
    sMap := make(map[string]byte)
    if len(p) != len(s) {
        return false
    }

    for i := 0; i < len(p); i++ {
        val, ok := pMap[p[i]]
        if ok {
            if val != s[i] {
                return false
            }
        } else {
            pMap[p[i]] = s[i]
        }
        val2, ok := sMap[s[i]]
        if ok {
            if val2 != p[i] {
                return false
            }
        } else {
            sMap[s[i]] = p[i]
        }
    }
    return true
}